algorithms/matrix/matrix_exponentiation.py

```diff
@@ -0,0 +1,60 @@
+def multiply(matA: list, matB: list) -> list:
+    """
+    Multiplies two square matrices matA and matB od size n x n
+    Time Complexity: O(n^3)
+    """
+    n = len(matA)
+    matC = [[0 for i in range(n)] for j in range(n)]
+
+    for i in range(n):
+        for j in range(n):
+            for k in range(n):
+                matC[i][j] += matA[i][k] * matB[k][j]
+
+    return matC
+
+def identity(n: int) -> list:
+    """
+    Returns the Identity matrix of size n x n
+    Time Complecity: O(n^2)
+    """
+    I = [[0 for i in range(n)] for j in range(n)]
+    
+    for i in range(n):
+        I[i][i] = 1
+    
+    return I
+
+def matrix_exponentiation(mat: list, n: int) -> list:
+    """
+    Calculates mat^n by repeated squaring
+    Time Complexity: O(d^3 log(n))
+                     d: dimesion of the square matrix mat
+                     n: power the matrix is raised to
+    """
+    if n == 0:
+        return identity(len(mat))
+    elif n % 2 == 1:
+        return multiply(matrix_exponentiation(mat, n - 1), mat)
+    else:
+        tmp = matrix_exponentiation(mat, n // 2)
+        return multiply(tmp, tmp)
+
+if __name__ == "__main__":
+    mat = [[1, 0, 2], [2, 1, 0], [0, 2, 1]]
+    
+    res0 = matrix_exponentiation(mat, 0)
```

Comment:
Add tests at [`tests/test_matrix.py`](https://github.com/keon/algorithms/blob/master/tests/test_matrix.py) similar to other tests.